### PR TITLE
Add GitHub Actions to build the project and to get notifications

### PR DESCRIPTION
**Pull request by rachelcarmena**
### Goal 

Add GitHub Actions to build the project.

The final list of checks would be:

* **Build**: GitHub Actions
* **Release**: Travis CI to Bintray
* **Documentation**: Bitrise

### Reason

There aren't notifications about failures with CI tasks.
Issue  #1678

### Note

This pull request doesn't change Travis CI because there is not an easy option to work with Oracle JDK from GitHub Actions (not convinced by the alternative of using a PPA).

### CODE REVIEW

**Comment by pakoito** [comment]
I'll need to resign from reviewing this because it's out of my area of expertise really. I'd check with @ffgiraldez if you can, because he stup a good chunk of it.

**Comment by rachelcarmena** [comment]
Thanks @pakoito !! Done, he'll review it on Monday when he's back :tada:

**Comment by rachelcarmena** [comment]
I've changed the scope of this pull request. Instead of a migration, adding a new GitHub action.

A new badge on `README.md` is also added.

**Comment by JorgeCastilloPrz** [comment]
Do we have a decision on moving the Arrow builds to GiHub actions then?

**Comment by rachelcarmena** [comment]
Thanks @JorgeCastilloPrz for the review!! You inspired me :wink:  I'm going to improve the labels and organization to make it clearer.

The idea:

* **Build**: GitHub Actions
* **Release**: Travis CI to Bintray
* **Documentation**: Bitrise

And I'm also going to remove the unnecessary duplicated checks when PRs (they run twice, for _push_ and _PR_ and I think it's enough for _push_).

**Comment by rachelcarmena** [comment]
BTW, we already had GitHub Actions in the "game". There is a `release-trigger` on master branch.

**Comment by rachelcarmena** [comment]
Done @JorgeCastilloPrz ! This is the final list:

* **Build**: GitHub Actions (all the branches)
* **Release**: Travis CI to Bintray (only `master` branch)
* **Documentation**: Bitrise (build for all the branches, deploy for `master` branch)

Though I changed `bitrise.yml`, it doesn't reflect the change here, so I think it's not taking into consideration that file. Maybe it's on Bitrise. 

I'm going to try to talk to @ffgiraldez

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      with:
+        java-version: 1.8
+        architecture: x64
+    - name: Build with Gradle
+      env:
+        JAVA_OPTS: -Xms512m -Xmx1024m
+      run: |
+        java -version
+        ./gradlew -v
+        ./gradlew clean build :arrow-benchmarks-fx:jmhClasses
```
[the remark this answers is not in the document]
Comment (rachelcarmena) [reply .github/workflows/build.yml]: Thanks @ffgiraldez for the review!! Good point :+1: ! I'm going to remove it for Travis CI check

[the remark this answers is not in the document]
Comment (rachelcarmena) [reply .github/workflows/build.yml]: Done!

**Comment by rachelcarmena** [comment]
> check how it's work and iterate over the time.

Ok!! :+1: 
 
> Points for further improvements:

I like all the points! Thanks @ffgiraldez for the review again :tada:

**Comment by rachelcarmena** [comment]
:point_right: This conversation continues on #1712.

**Pull request merged**